### ISSUE TITLE
Metadata editing in pod5 files

### ISSUE THREAD

**Issue by DCossey** [nanoporetech/pod5-file-format#100]
Hi, we were running into an error during dorado basecalling (of recovered files after a failed run) due to incorrect sequencing kit: 

[2023-12-15 14:35:04.358] [error] Unknown sequencing_kit: FLO-PRO114M

So then we checked our pod5 files and saw the following:	

flow_cell_product_code: FLO-PRO114M
	sequencing_kit: FLO-PRO114M

Is it possible to edit the incorrect sequencing kit somehow?

**Comment by HalfPhoton** [comment]
Hi @DCossey ,
Yes it is possible to fix your metadata although it's not particularly clean as pod5 files are immutable.

There is a short part of the [documentation](https://pod5-file-format.readthedocs.io/en/latest/docs/api.html#adding-reads-example) relating to this, but here's a snippet more tailored to your issue. 
You need to edit the [RunInfo](https://pod5-file-format.readthedocs.io/en/latest/reference/api/pod5.pod5_types.html#pod5.pod5_types.RunInfo) of each read.

```python
import pod5

# New output file for edited data
with pod5.Writer("output.pod5") as writer:
    # Read all records
    with pod5.Reader("input.pod5") as reader:
          # Iterate over immutable ReadRecords
          for record in reader:
               # Convert to mutable Read
               read = record.to_read()
               # Edit the value
               read.run_info.sequencing_kit = "sequencing_kit_here"
               # Write the edited read
               writer.add_read(read)
``` 

Kind regards,
Rich

**Comment by jennieli421** [comment]
I followed the example code and revised a pod5 file successfully. However, when I try to check the content using `pod5 view`, I get this error (the same error if I try to view an untouched pod5): 
```
POD5 has encountered an error: 'Error while processing "output2.pod5''

For detailed information set POD5_DEBUG=1'
```

**Comment by HalfPhoton** [comment]
What command are you running?

**Comment by jennieli421** [comment]
```
pod5 view "output2.pod5"
```

**Comment by HalfPhoton** [comment]
Can you try without the quotes please?

**Comment by jennieli421** [comment]
Tried and still the same error.

**Comment by HalfPhoton** [comment]
Can you run the following then:
```python
pod5 --version
POD5_DEBUG=1 pod5 view output2.pod5
```
And then share the contents of the pod5 `.log` files that are generated?

**Comment by jennieli421** [comment]
Yes my `polars==0.20`. Note that I have to run `pip install -U polars==0.19`, otherwise it would say "requirement already satisfied`. The error was fixed. Thanks!

**Comment by HalfPhoton** [comment]
Fantastic, sorry about that last issue - we're patching this as we speak

**Comment by zacksaud** [comment]
> Hi @DCossey , Yes it is possible to fix your metadata although it's not particularly clean as pod5 files are immutable.
> 
> There is a short part of the [documentation](https://pod5-file-format.readthedocs.io/en/latest/docs/api.html#adding-reads-example) relating to this, but here's a snippet more tailored to your issue. You need to edit the [RunInfo](https://pod5-file-format.readthedocs.io/en/latest/reference/api/pod5.pod5_types.html#pod5.pod5_types.RunInfo) of each read.
> 
> ```python
> import pod5
> 
> # New output file for edited data
> with pod5.Writer("output.pod5") as writer:
>     # Read all records
>     with pod5.Reader("input.pod5") as reader:
>           # Iterate over immutable ReadRecords
>           for record in reader:
>                # Convert to mutable Read
>                read = record.to_read()
>                # Edit the value
>                read.run_info.sequencing_kit = "sequencing_kit_here"
>                # Write the edited read
>                writer.add_read(read)
> ```
> 
> Kind regards, Rich

How would I run this script from ubuntu? I tried running- $python3 pod5editscript.py data.pod5
but it says "ModuleNotFoundError: No module named 'pod5'"

I know this is an extremely basic question but I an extremely low-level bioinformatician.

Many thanks in advance
